src/mpid/ch4/include/mpidpost.h

```diff
@@ -9,6 +9,17 @@
 #include "mpir_datatype.h"
 #include "mpidch4.h"
 
+MPL_STATIC_INLINE_PREFIX MPIR_Request *MPID_Request_create_from_comm(MPIR_Request_kind_t kind,
+                                                                     MPIR_Comm * comm)
+{
+    MPIR_Request *req;
+    int vci = MPIDI_get_comm_vci(comm);
```

Comment:
My only question is whether we'd be better off exposing `MPID_get_comm_vci` and implement the request creation logic in `MPIR` instead? Functionality-wise this makes sense.

Reply:
The issue is ch3. If we move vci to `MPIR`, we would require all devices to support VCI. Until most vendors adopt VCI, I'm afraid it will create too many interruptions to downstream vendors.

Reply:
I definitely don't want to move VCIs to MPIR. If request pool is the only thing associated with a VCI, then I guess it shouldn't matter whether we expose `MPID_get_comm_pool` or `MPID_Request_create_from_comm`. Is there anything else in MPIR that is pool-associated? Probably not...

Reply:
The request pool ideally should be hidden inside ch4 as well, except the handle allocation logic is too integrated in the MPIR-layer. Thus, we should try only access the request pool interface from within ch4.